astroquery/mast/collections.py

```diff
@@ -109,10 +110,18 @@ def query_region_async(self, coordinates, *, radius=0.2*u.deg, catalog="Hsc",
                   'dec': coordinates.dec.deg,
                   'radius': radius.deg}
 
+        # valid criteria keywords
+        valid_criteria = []
+
         # Determine API connection and service name
         if catalog.lower() in self._service_api_connection.SERVICES:
             self._current_connection = self._service_api_connection
             service = catalog
+
+            # adding additional user specified parameters
+            for prop, value in criteria.items():
```

Comment:
Any chance to validate these, or there is way to large a scatter in what's allowed for each of these catalogs?

Reply:
So, the only service that runs through Catalogs.MAST (currently), and thus the service API connection, is PanSTARRS. I added `Catalogs._get_service_col_config` and `Catalogs._validate_service_criteria` to fetch the column metadata for PanSTARRS and check parameters.